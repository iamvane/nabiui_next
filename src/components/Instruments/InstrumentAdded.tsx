import * as React from 'react';
import {
  Grid,
  Icon,
  IconButton,
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
} from '@material-ui/core';

import { InstrumentAddedComponent } from './constants';

interface Props {
  instrument: string;
  skillLevel: string;
  deleteInstrument: (instrumentName: string) => void;
}

const InstrumentAdded: React.StatelessComponent<Props> = props => {
  const instrumentNameUppercase = props.instrument && 
    props.instrument.charAt(0).toUpperCase() + props.instrument.slice(1);

  return (
    <Grid item={true} md={6} xs={12}>
      <ListItem>
        <ListItemText
          primary={
            InstrumentAddedComponent.instrumentAdded.replace(
              InstrumentAddedComponent.instrumentPlaceholder,
              instrumentNameUppercase
            ).replace(
              InstrumentAddedComponent.skillLevelPlaceholder,
              props.skillLevel
            )
          }
        />
        <ListItemSecondaryAction>
          <IconButton aria-label="Delete" onClick={() => props.deleteInstrument(props.instrument)}>
            <Icon>delete</Icon>
          </IconButton>
        </ListItemSecondaryAction>
      </ListItem>
    </Grid>
  );
};

export default InstrumentAdded;
