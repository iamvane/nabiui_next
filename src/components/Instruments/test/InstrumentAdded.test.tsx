import * as React from 'react';
import InstrumentAdded from 'components/Instruments/InstrumentAdded';
import { shallow } from 'enzyme';

describe('InstrumentAdded', () => {
  let wrapper: any;
  const deleteInstrument: (instrumentName: string) => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <InstrumentAdded
          instrument=""
          skillLevel=""
          deleteInstrument={deleteInstrument}
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
