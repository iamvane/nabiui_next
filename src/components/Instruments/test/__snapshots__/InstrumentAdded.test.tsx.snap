// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentAdded Shallow Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InstrumentAdded
    deleteInstrument={[MockFunction]}
    instrument=""
    skillLevel=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="-  ()"
        />
        <WithStyles(ListItemSecondaryAction)>
          <WithStyles(IconButton)
            aria-label="Delete"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              delete
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>,
      "item": true,
      "md": 6,
      "xs": 12,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItemText)
            primary="-  ()"
          />,
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "primary": "-  ()",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "delete",
              },
              "ref": null,
              "rendered": "delete",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(ListItem)>
          <WithStyles(ListItemText)
            primary="-  ()"
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>,
        "item": true,
        "md": 6,
        "xs": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItemText)
              primary="-  ()"
            />,
            <WithStyles(ListItemSecondaryAction)>
              <WithStyles(IconButton)
                aria-label="Delete"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </WithStyles(ListItemSecondaryAction)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "primary": "-  ()",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                aria-label="Delete"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "delete",
                },
                "ref": null,
                "rendered": "delete",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
